src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ExpressionCompilerTestBase.cs

```diff
@@ -385,22 +385,41 @@ internal static Alias Alias(DkmClrAliasKind kind, string name, string fullName,
 
             return MethodDebugInfo<TypeSymbol, LocalSymbol>.ReadMethodDebugInfo((ISymUnmanagedReader3)symReader, symbolProvider, MetadataTokens.GetToken(peMethod.Handle), methodVersion: 1, ilOffset: ilOffset, isVisualBasicMethod: false);
         }
-
-        internal static SynthesizedAttributeData GetDynamicAttributeIfAny(IMethodSymbol method)
+        
+        internal static void CheckAttribute(IEnumerable<byte> assembly, IMethodSymbol method, AttributeDescription description, bool expected)
         {
-            return GetAttributeIfAny(method, "System.Runtime.CompilerServices.DynamicAttribute");
-        }
+            var module = AssemblyMetadata.CreateFromImage(assembly).GetModules().Single().Module;
 
-        internal static SynthesizedAttributeData GetTupleElementNamesAttributeIfAny(IMethodSymbol method)
-        {
-            return GetAttributeIfAny(method, "System.Runtime.CompilerServices.TupleElementNamesAttribute");
-        }
+            var typeHandle = module
+                .GroupTypesByNamespaceOrThrow(StringComparer.Ordinal)
+                .SelectMany(group => group)
```

Comment:
>.SelectMany(group => group) [](start = 16, length = 27)

This feels too complicated. Consider simply using module.MetadataReader.TypeDefinitions collection dirtectly.